generic/ltp.py

```diff
@@ -130,6 +149,52 @@ def setUp(self):
         build.make(ltp_dir)
         build.make(ltp_dir, extra_args='install')
 
+        ltp_tstnet_dir = os.path.join(self.ltpbin_dir, 'testcases/bin/tst_net.sh')
+        if not os.path.exists(ltp_tstnet_dir):
+            self.log.info("File tst_net.sh does not exist")
+
+        # modifying input parameters in tst_net.sh file to run ltp on public address instead of localhost
+        filehandle = open(ltp_tstnet_dir, "r")
+        listOfLines = filehandle.readlines()
+        filehandle.close()
+
+        lineNo = [149, 150, 210, 211, 212, 988, 989, 990, 991, 1081, 1082, 1086, 1087]
+        newline = ["        LHOST_IFACES=\"${LHOST_IFACES:-" + self.host_iface + "}\"\n", "        RHOST_IFACES=\"${RHOST_IFACES:-" + self.peer_iface + "}\"\n", "        tst_require_cmds ssh\n", "        use=\"SSH\"\n", "        rcmd=\"ssh -q " + self.peer_user + "@" + self.peer_public_ip + "\"\n", "IPV4_LHOST=\"${IPV4_LHOST:-" + self.host_ipv4 + "/" + str(self.prefix) + "}\"\n", "IPV4_RHOST=\"${IPV4_RHOST:-" + self.peer_ipv4 + "/" + str(self.prefix) + "}\"\n", "IPV6_LHOST=\"${IPV6_LHOST:-" + self.host_ipv6 + "}\"\n", "IPV6_RHOST=\"${IPV6_RHOST:-" + self.peer_ipv6 + "}\"\n", "export LHOST_IFACES=\"${LHOST_IFACES:-" + self.host_iface + "}\"\n", "export RHOST_IFACES=\"${RHOST_IFACES:-" + self.peer_iface + "}\"\n", "export LHOST_HWADDRS=\"${LHOST_HWADDRS:-" + self.host_hwaddr + "}\"\n", "export RHOST_HWADDRS=\"${RHOST_HWADDRS:-" + self.peer_hwaddr + "}\"\n"]
+
+        for i, val in enumerate(lineNo):
+            listOfLines[val] = newline[i]
+            filehandle = open(ltp_tstnet_dir, "w")
+            filehandle.writelines(listOfLines)
+
+        filehandle.seek(0)
+        filehandle.truncate()
+        for i, line in enumerate(listOfLines):
+            if i not in [206, 207, 208, 209, 213]:
+                filehandle.write(line)
```

Comment:
applying changes patch is a better way like we have here : https://github.com/avocado-framework-tests/avocado-misc-tests/blob/4c6bfa2d9ece6de52a883a6cc2e1c195c1161eaf/io/disk/iozone.py#L456

Reply:
@abdhaleegit This is used to apply patch , but what i am doing here is deleting some lines from the tst_net.sh file 
if at all i have to use this apply patch thing, i need to create a copy of file from ltp tool, modify the file and then apply the file again as a patch. Which increases no of lines in the code.

If at all i have a file then i can keep the copy of file and apply it every time. but here in every test case ltp is installed and configured then the tst_net.sh file is generated.

Please let me know keeping a copy of tst_net.sh script is a right way to do. If yes then where the file can be stored ?

Reply:
may be after ltp installed apply the patch

Reply:
In short, apply patch and managing patch is easy comparing to line numbers... if in future there is script change ..than you have to go to each line and get the numbers.. but here in patch case.. just take a diff every time there is change ..so easy

Reply:
regarding keeping a sh file.. we can keep it in data folder.. but again .. there should be a way to apply patch for every new installation of ltp

Reply:
@abdhaleegit @PraveenPenguin  Applying a patch would be difficult and also not possible here as i am not having the static data in the patch that needs to be modified in the file

The file i am modifying is with the input values which i am fetching from yaml file, like interface names, ip addresses, netmask values, ssh login details etc. 
The values which i am modifying in tsh_net.sh file will be different for different setups, as it will be fetched from yaml input file. Hence i am keeping it same what is done for now. Which makes it easy to apply and working.